l2os/drivers/l2output/driver.go

```diff
@@ -206,6 +206,15 @@ func (d *Driver) CraftTx(
 		return nil, fmt.Errorf("error resolving checkpoint block: %v", err)
 	}
 
+	l2Header, err := d.cfg.L2Client.HeaderByNumber(ctx, nextCheckpointBlock)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving checkpoint block: %v", err)
+	}
```

Comment:
Why not change the `outputRootAtBlock` RPC in the rollup-node to serve the L2 block header. That way it's always consistent. The l2 output root submitter shouldn't even need to directly talk to a L2 geth node I think

Reply:
That wouldn't solve the original issue nor a race condition on the proof. Original is that the `timestamp -> block number` conversion was wrong. The race condition is that we get by block number but don't check the proof against the header - the only way to do this is by checking the state root.

Reply:
FWIW, the server of the output root already does get a header and verifies the proof against the state root: https://github.com/ethereum-optimism/optimistic-specs/blob/5189b9a26831f0a3e51587e3ddb9aba65195fdaf/opnode/node/api.go#L73
It would be nice to serve the header in the same RPC response, and then the timestamp / block number can be used without extra calls with a separate RPC client to L2 geth.